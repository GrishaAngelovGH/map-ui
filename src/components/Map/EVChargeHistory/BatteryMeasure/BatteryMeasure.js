const BatteryMeasure = ({ measure }) => (
  <div className='row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow'>
    <div className='col-md-2'>
      <i className='bi bi-car-front fs-1'></i>
    </div>
    <div className='col-md-7'>
      <p className='m-0 fw-bold'>{measure.location.city}</p>
      <p className='m-0 text-secondary'>{measure.location.city}, {measure.location.country}</p>
    </div>
    <div className='col-md-3 d-flex align-items-center'>
      <i className={`bi bi-${measure.battery < 100 ? 'battery-half' : 'battery-full'} fs-1`}></i>
      <span className='fw-bold mx-2'>{measure.battery}%</span>
    </div>
  </div>
)

export default BatteryMeasure