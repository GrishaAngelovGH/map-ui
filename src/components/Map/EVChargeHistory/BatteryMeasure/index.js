import BatteryMeasure from './BatteryMeasure'

export default BatteryMeasure