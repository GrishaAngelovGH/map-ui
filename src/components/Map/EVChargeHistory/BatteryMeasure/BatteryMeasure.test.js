import { render } from '@testing-library/react'

import BatteryMeasure from './BatteryMeasure'

test('should render BatteryMeasure component with full battery charge', () => {
  const view = render(
    <BatteryMeasure
      measure={{
        location: { city: 'London', country: 'England, UK' },
        battery: 100
      }}
    />
  )

  expect(view).toMatchSnapshot()
})

test('should render BatteryMeasure component with lower battery charge', () => {
  const view = render(
    <BatteryMeasure
      measure={{
        location: { city: 'London', country: 'England, UK' },
        battery: 65
      }}
    />
  )

  expect(view).toMatchSnapshot()
})