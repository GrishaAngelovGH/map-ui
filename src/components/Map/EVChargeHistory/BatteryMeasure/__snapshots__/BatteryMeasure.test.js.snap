// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BatteryMeasure component with full battery charge 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
      >
        <div
          class="col-md-2"
        >
          <i
            class="bi bi-car-front fs-1"
          />
        </div>
        <div
          class="col-md-7"
        >
          <p
            class="m-0 fw-bold"
          >
            London
          </p>
          <p
            class="m-0 text-secondary"
          >
            London
            , 
            England, UK
          </p>
        </div>
        <div
          class="col-md-3 d-flex align-items-center"
        >
          <i
            class="bi bi-battery-full fs-1"
          />
          <span
            class="fw-bold mx-2"
          >
            100
            %
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
    >
      <div
        class="col-md-2"
      >
        <i
          class="bi bi-car-front fs-1"
        />
      </div>
      <div
        class="col-md-7"
      >
        <p
          class="m-0 fw-bold"
        >
          London
        </p>
        <p
          class="m-0 text-secondary"
        >
          London
          , 
          England, UK
        </p>
      </div>
      <div
        class="col-md-3 d-flex align-items-center"
      >
        <i
          class="bi bi-battery-full fs-1"
        />
        <span
          class="fw-bold mx-2"
        >
          100
          %
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render BatteryMeasure component with lower battery charge 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
      >
        <div
          class="col-md-2"
        >
          <i
            class="bi bi-car-front fs-1"
          />
        </div>
        <div
          class="col-md-7"
        >
          <p
            class="m-0 fw-bold"
          >
            London
          </p>
          <p
            class="m-0 text-secondary"
          >
            London
            , 
            England, UK
          </p>
        </div>
        <div
          class="col-md-3 d-flex align-items-center"
        >
          <i
            class="bi bi-battery-half fs-1"
          />
          <span
            class="fw-bold mx-2"
          >
            65
            %
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mb-4 justify-content-evenly align-items-center border rounded p-4 shadow"
    >
      <div
        class="col-md-2"
      >
        <i
          class="bi bi-car-front fs-1"
        />
      </div>
      <div
        class="col-md-7"
      >
        <p
          class="m-0 fw-bold"
        >
          London
        </p>
        <p
          class="m-0 text-secondary"
        >
          London
          , 
          England, UK
        </p>
      </div>
      <div
        class="col-md-3 d-flex align-items-center"
      >
        <i
          class="bi bi-battery-half fs-1"
        />
        <span
          class="fw-bold mx-2"
        >
          65
          %
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
